plugins/plugin-gdb/che-plugin-gdb-ide/src/main/java/org/eclipse/che/plugin/gdb/ide/configuration/GdbConfigurationPagePresenter.java

```diff
@@ -116,24 +119,38 @@ public void go(AcceptsOneWidget container) {
     }
 
     private void setHostsList() {
-        machineServiceClient.getMachines(appContext.getWorkspaceId()).then(new Operation<List<MachineDto>>() {
-            @Override
-            public void apply(List<MachineDto> machines) throws OperationException {
-                @SuppressWarnings("unchecked")
-                Promise<RecipeDescriptor>[] recipePromises = (Promise<RecipeDescriptor>[])new Promise[machines.size()];
-
-                for (int i = 0; i < machines.size(); i++) {
-                    String location = machines.get(i).getConfig().getSource().getLocation();
-                    String recipeId = getRecipeId(location);
-                    recipePromises[i] = recipeServiceClient.getRecipe(recipeId);
-                }
+        List<Machine> machines = getMachines();
+        if (machines.isEmpty()) {
+            return;
+        }
+
+        @SuppressWarnings("unchecked")
+        Promise<RecipeDescriptor>[] recipePromises = (Promise<RecipeDescriptor>[])new Promise[machines.size()];
```

Comment:
I think we should use collections instead of arrays where ever it is possible and simply use [`toArray`](https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#toArray-T:A-) method where we really need arrays.